Mathlib/Computability/ContextFreeGrammar.lean

```diff
@@ -162,9 +166,17 @@ lemma Derives.eq_or_head {u w : List (Symbol T g.NT)} (huw : g.Derives u w) :
     u = w ∨ ∃ v : List (Symbol T g.NT), g.Produces u v ∧ g.Derives v w :=
   Relation.ReflTransGen.cases_head huw
 
+lemma Derives.iff_eq_or_head {u w : List (Symbol T g.NT)} :
+    g.Derives u w ↔ u = w ∨ ∃ v : List (Symbol T g.NT), g.Produces u v ∧ g.Derives v w :=
+  Relation.ReflTransGen.cases_head_iff
+
 lemma Derives.eq_or_tail {u w : List (Symbol T g.NT)} (huw : g.Derives u w) :
-    u = w ∨ ∃ v : List (Symbol T g.NT), g.Derives u v ∧ g.Produces v w :=
```

Comment:
Note: I switched eq_or_tail from `u = w` to `w = u`: it better follows the convention of `ReflTransGen.cases_tail`. I have little idea why the original PR deviated from this.

Reply:
Thanks!